clip/README.md

```diff
@@ -0,0 +1,104 @@
+# (OpenAI) CLIP
+
+An example of visual-language representation learning using MLX.
+
+CLIP, for Contrastive Language-Image Pre-training, is a powerful representation learning method that aims to learn shared embedding space for both images and text.
+
+Paper: https://arxiv.org/pdf/2103.00020.pdf
+
+### Setup
+
+Install the dependencies:
+
+```
+pip install -r requirements.txt
+```
+
+Next, download the model configuration and weights. To download the model, it is necessary to specify the desired HuggingFace configuration and checkpoint. In this example, we will use ```
+openai/clip-vit-base-patch32```.
+
+```
+python download.py openai/clip-vit-base-patch32
+```
+If the download is successful, the folder ```weights``` will appear. In the ```weights``` folder, there will be the HuggingFace configuration and weights for ```openai/clip-vit-base-patch32```.
+
+Now, convert the PyTorch weights to MLX with:
+
+```
+python convert.py --torch-path weights/openai/clip-vit-base-patch32 --mlx-path weights/mlx/openai/clip-vit-base-patch32
+```
```

Comment:
Why not do this in one step instead of two?

Reply:
I did not know that `snapshot_download` exists. There is no reason why we should use two steps instead of one.

Reply:
Let's use the one step.